subcommands/version/updates.go

```diff
@@ -91,6 +99,72 @@ func (f *FioctlUpdateFinder) FindLatest() (*FioctlUpdate, error) {
 	return update, nil
 }
 
+type progressBar struct {
+	total   int64
+	written int64
+	sha     hash.Hash
+	buff    bytes.Buffer
+}
+
+func (p *progressBar) Write(buff []byte) (n int, err error) {
+	if _, err = p.sha.Write(buff); err != nil {
+		return 0, err
+	}
+	n, err = p.buff.Write(buff)
```

Comment:
Can we write immediately into a temporary file instead?
Buffering 20MB into memory does not seem sane enough.

Reply:
we can do a temp file. i tried to avoid because it makes the code a bit more cumbersome and i also was a bit apprehensive on storing something, even temporary, to disk that might be hostile/unverified content.

Its easy enough though - especially now that I've worked mentally through whats required of the windows updater logic.

Reply:
There are pros and cons each way.

But, there are code paths where we can write garbage to disk anyway.
So, it's better to have _just one evil_ than a duo.

You've raised a good point about writing _garbage that can be potentially executed_.
So, I'd do this:
```
tmpName := fileName + ".tmp"
tmp = os.Open(tmpName, "w", "600")
// write and verify here
if err != nil {
  os.Remove(tmpName)
}
os.Chmod(tmpName, st.Mode())
os.Rename(tmpName, fileName)
```

That way we protect the user from both _over-consuming computer memory_ and _running unverified code_.

Reply:
I think that keeping 20M in RAM is not an issue at all since modern PCs/laptops have at least 4GB.

Reply:
I just would do it in the way that is the most simple yet reliable.

Reply:
the more files/chmod/renames we have the more `if err!=nil` bits you get. I think volodya's requires to just write it straight to disk is fine. ironically - the idea to write it to `/tmp` - that's *RAM* on many linux systems.

I'm going to squash my commits and push something clean. Then I'll do this change so that it won't create crazy conflicts

Reply:
> That way we protect the user from both over-consuming computer memory and running unverified code.

NOTE: If you at how LInux's VFS layer works - this whole file is going to be in memory. And we limit how much we are willing to read based on TUF.

I think the decision driver here is what looks good stylistically and is easy to maintain. That said - writing to a file doesn't change it much - so i'm going to do that.

Reply:
probably easier to read the new Update() logic as a whole, but here's the delta 1f151acebe8e10287b6ba697d9e24137827762dc

I found a slightly cleaner CLI to run on Win32 while testing